ledger/ledgercore/statedelta.go

```diff
@@ -70,12 +70,24 @@ type StateDelta struct {
 	PrevTimestamp int64
 }
 
+// HoldingAction is an enum of actions on holdings
+type HoldingAction uint64
+
+const (
+	// ActionCreate is for asset holding creation
+	ActionCreate HoldingAction = 1 + iota
+	// ActionDelete is for asset holding creation
+	ActionDelete
+)
+
 // AccountDeltas stores ordered accounts and allows fast lookup by address
 type AccountDeltas struct {
 	// actual data
 	accts []basics.BalanceRecord
 	// cache for addr to deltas index resolution
 	acctsCache map[basics.Address]int
+	// holdings keeps track of created and deleted holdings per address
+	holdings map[basics.Address]map[basics.AssetIndex]HoldingAction
```

Comment:
I can't see here were you're using this `holding` structure, but it looks here as we have some first-level duplication here : the data inside this map should go into the `accts`.

Reply:
this map tracks created and **deleted** holdings. See `SetHoldingDelta` below.

Reply:
yes, I realize that.. I'm wondering why wound't we add this map to the `PersistedBalanceRecord` instead ?

Reply:
It is possible but `PersistedBalanceRecord` is also used in eval for accounts prefetching and in that context it does not need holdings delta at all. I'll revisit your suggestion as move forward with params implementation.